cachi2/core/models/output.py

```diff
@@ -48,7 +49,29 @@ def resolve_value(self, mappings: Dict[str, str]) -> str:
             log.debug(f"Adjusting a legacy path variable value '{self.name}:{self.value}'")
             self.value = "${output_dir}/" + self.value
 
-        return string.Template(self.value).safe_substitute(mappings)
+        # "Recursively" resolve potentially nested variables up to len(mappings) tries
+        ret = self.value
+        placeholders_seen = []
+        for i, m in enumerate(mappings):
+            log.debug(f"EnvironmentVariable resolution iteration {i}.: {ret}")
+
+            placeholders_seen.append(m)
```

Comment:
The nested variable resolution is very cool :+1: 

But in the current state, cycle detection doesn't work correctly (depends on the order of the `mapping`)
```python
>>> e = EnvironmentVariable(name="A", value="$C")
>>> e.resolve_value({"C": "$B", "B": "1"})
'1'
>>> e.resolve_value({"B": "1", "C": "$B"})
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/acmiel/RedHat/Cachito/cachi2/cachi2/core/models/output.py", line 66, in resolve_value
    raise Cachi2Error(
cachi2.core.errors.Cachi2Error: Detected a cycle in environment variable expansion of 'A'
```

Reply:
Doh! Nasty bug, shameful...@chmeliik thanks for pointing this out!

Reply:
I know you'll try to solve this meticulously first thing in the morning, but I wonder if we need nested variable resolution in the first place. I does look nice, but it adds quite some complexity, and maybe we don't need it right now.

Our current use case can be solved if we declare GOPROXY as:
```
file://${output_dir}/deps/gomod/pkg/mod/cache/download
```
... and we can hold on adding nested variable resolution for when a use case that can't be solved this way comes up.

Reply:
@chmeliik updated with IMO a more robust solution, see if you can spot another resolution problem. I also tweaked and rearranged the test data so that your finding is covered.

Reply:
@brunoapimentel I understand the motivation and what you suggest was my initial approach, but then I figured that I hate maintaining duplicate static strings which leads to the need to fix them eventually at all places. It all comes down to immediate effect vs hardening. IMO it is the immediate impact fixes that usually lead to and add up to the technical debt that will likely never be addressed, but there surely is a line to be drawn. If you feel strongly about the added complexity (which won't even be exercised 99% of the cases) being too high, then I'll drop the change no problem.

Reply:
I don't feel strongly about it, my reasoning was mostly to avoid extra work, since we will need to reach a reasonable solution with this nested variable resolution algorithm.